### PR TITLE
Have dolphin_scmrev run `ScmRevGen` from `CMAKE_CURRENT_SOURCE_DIR`

### PR DESCRIPTION
**Pull request by JesseTG**
This is another little fix to simplify `FetchContent`'ing Dolphin. Without this PR, the `dolphin_scmrev` target will try to use a file named `ScmRevGen.cmake` in the _including_ project's `CMake` directory.

### CODE REVIEW

**Comment by AdmiralCurtiss** [comment]
Makes sense. There's a few other instances of `CMAKE_SOURCE_DIR` that probably want to execute relative to the current directory instead of relative of the top-level CMake file, too, such as 

https://github.com/dolphin-emu/dolphin/blob/acb18a58cf77130592d06bd6c7841a2fb05b8996/CMakeLists.txt#L158-L160

**Comment by JesseTG** [comment]
> Makes sense. There's a few other instances of `CMAKE_SOURCE_DIR` that probably want to execute relative to the current directory instead of relative of the top-level CMake file, too, such as
> 
> https://github.com/dolphin-emu/dolphin/blob/acb18a58cf77130592d06bd6c7841a2fb05b8996/CMakeLists.txt#L158-L160

Sure, I can knock 'em out for you if you'd like. I'll comb through myself, but are there any instances you'd like to highlight?

**Comment by JesseTG** [comment]
@AdmiralCurtiss How's this?

File: Source/Android/jni/CMakeLists.txt [hunk 1]
```diff
@@ -55,17 +55,17 @@ PRIVATE
 
 target_include_directories(main
 PRIVATE
-    ${CMAKE_SOURCE_DIR}/Externals/libadrenotools/include
-    ${CMAKE_SOURCE_DIR}/Externals/VulkanMemoryAllocator/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/Externals/libadrenotools/include
+    ${CMAKE_CURRENT_SOURCE_DIR}/Externals/VulkanMemoryAllocator/include
```
Comment (AdmiralCurtiss) [inline Source/Android/jni/CMakeLists.txt:59]: This now points at `[dolphin]/Source/Android/jni/Externalslibadrenotools/include`, and so on.

Comment (JesseTG) [reply Source/Android/jni/CMakeLists.txt:59]: Okay, I'll revert my larger set of changes and keep the scope down.

**Comment by AdmiralCurtiss** [comment]
I'm guessing you will be back but yeah sure this is fine.

**Comment by JesseTG** [comment]
> I'm guessing you will be back but yeah sure this is fine.

Yes, this is a safe bet. Thanks for merging this. :+1:

**Pull request merged**